### PR TITLE
Genuary: include linked media from Genuary threads

### LINKED ISSUE #769: Genuary page does not include linked media from Genuary threads
### What happened?

The pinned messages of each genuary thread are automatically checked for media, but the process does not consider that submissions may be embedded links (cdn.discordapp, imgur.com, etc) to media (mp4, gif, etc) rather than a direct Discord attachment. As a result, some Genuary submissions are missing from the [Genuary page](https://acmcsuf.com/genuary/2023).

### PR DESCRIPTION
**Pull request by jiink**
To solve #769, the logic to get media from a pinned message needs to be changed.

### CODE REVIEW

File: scripts/transform-genuary/in.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          "content_type": "video/quicktime",
+          "size": 29223341,
+          "url": "https://cdn.discordapp.com/attachments/1055680245740605450/1061515218653548564/Screen_Recording_2023-01-07_at_9.20.04_PM.mov",
+          "proxy_url": "https://media.discordapp.net/attachments/1055680245740605450/1061515218653548564/Screen_Recording_2023-01-07_at_9.20.04_PM.mov",
+          "height": 1454,
+          "width": 2222
+        }
+      ]
+    },
+
```
Comment (EthanThatOneKid) [inline scripts/transform-genuary/in.json]: Nit: Rogue empty line.

File: scripts/transform-genuary/test-data.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          "content_type": "video/quicktime",
+          "size": 29223341,
+          "url": "https://cdn.discordapp.com/attachments/1055680245740605450/1061515218653548564/Screen_Recording_2023-01-07_at_9.20.04_PM.mov",
+          "proxy_url": "https://media.discordapp.net/attachments/1055680245740605450/1061515218653548564/Screen_Recording_2023-01-07_at_9.20.04_PM.mov",
+          "height": 1454,
+          "width": 2222
+        }
+      ]
+    },
+
```
Comment (EthanThatOneKid) [inline scripts/transform-genuary/test-data.json]: Nit: Rogue empty line.

File: scripts/transform-genuary/in.json (hunk 1 above)
Comment (EthanThatOneKid) [reply scripts/transform-genuary/in.json]: Resolved in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/821/commits/034a277d3a450c1ef240c226f96fb4029c5c4375>.

File: scripts/transform-genuary/test-data.json (hunk 2 above)
Comment (EthanThatOneKid) [reply scripts/transform-genuary/test-data.json]: Resolved in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/821/commits/034a277d3a450c1ef240c226f96fb4029c5c4375>.

**Review by EthanThatOneKid — APPROVED** [review]
Congratulations on your amazing PR, @jiink!

Excited to finally see `.github/workflows/update_genuary.yaml` trigger in prod!

**Review by EthanThatOneKid — CHANGES_REQUESTED** [review]
Oh no!!

image

File: .github/workflows/check_website.yaml [hunk 3]
```diff
@@ -63,3 +63,15 @@ jobs:
...
       - name: Test Dependency Size
         run: npm run test:size
+
+      - name: Transform Genuary Check Changes
+        id: transform-genuary-check-changes
+        run: if git diff --exit-code --quiet -- scripts/transform-genuary; then
+          echo "updated=0" >> $GITHUB_OUTPUT
+          else
+          echo "updated=1" >> $GITHUB_OUTPUT
+          fi
```
Comment (EthanThatOneKid) [inline .github/workflows/check_website.yaml]: Apologies, I missed this syntax error in my previous review. This text is multiline.

```suggestion
        run: |
           if git diff --exit-code --quiet scripts/transform-genuary; then
             echo "updated=0" >> $GITHUB_OUTPUT;
           else
             echo "updated=1" >> $GITHUB_OUTPUT;
           fi
```

**Review by EthanThatOneKid — CHANGES_REQUESTED** [review]

**Review by EthanThatOneKid — APPROVED** [review]

**Comment by EthanThatOneKid** [comment]
Keep an eye out for <https://acmcsuf.com/code/actions/workflows/update_genuary.yaml> on March 29th, 2023! 👀

**Review by jiink — COMMENTED** [review]
Looks good!

**Pull request merged**